### PR TITLE
fix: return unformatted if object

### PR DESCRIPTION
**Pull request by tinytim84**

### CODE REVIEW

**Review by AndrewLeedham — CHANGES_REQUESTED** [review]
The prettier implementation does not need to change, but it expects a string. The problem is here: https://github.com/xeroxinteractive/fractal-theme/blob/dev/views/partials/browser/panel-context.nunj#L14 prettier is running before format, but format is the filter that converts it to a string, they need reversing. Also while you are in there, prettier should be added to the non-collated part: https://github.com/xeroxinteractive/fractal-theme/blob/dev/views/partials/browser/panel-context.nunj#L5

**Comment by tinytim84** [comment]
This should be ready for review now, there are still some test bugs in the corresponding [framework PR](https://github.com/xeroxinteractive/xerox-framework/pull/1520), but this one is ready to be reviewed while I fix those.

**Comment by AndrewLeedham** [comment]
The text seems to move when clicking between them can that be removed? Seems to be because neither is selected during loading, I think 1 should always be selected and the contents should have a loading animation.

Also the 0.6 transparency for the loading state looks strange when it loads so quick. There is already a loading animation for when a component is rendering on the server can we use that if the Ajax call takes more than 100ms, otherwise just swap straight away?

Fractal Theme Jumpyness

**Review by AndrewLeedham — CHANGES_REQUESTED** [review]
The text seems to move when clicking between them can that be removed? Seems to be because neither is selected during loading, I think 1 should always be selected and the contents should have a loading animation.

Also the 0.6 transparency for the loading state looks strange when it loads so quick. There is already a loading animation for when a component is rendering on the server can we use that if the Ajax call takes more than 100ms, otherwise just swap straight away?

Fractal Theme Jumpyness

**Comment by tinytim84** [comment]
> The text seems to move when clicking between them can that be removed? Seems to be because neither is selected during loading, I think 1 should always be selected and the contents should have a loading animation.
> 
> Also the 0.6 transparency for the loading state looks strange when it loads so quick. There is already a loading animation for when a component is rendering on the server can we use that if the Ajax call takes more than 100ms, otherwise just swap straight away?
> 
> Fractal Theme Jumpyness

I've done these updates.  Your machine must be much faster than mine, I never have that short a delay between changing variants!

File: assets/js/theme.js [hunk 1]
```diff
@@ -44,11 +46,72 @@ if (frctl.env === 'server') {
...
+
+function switchVariant(clickedVariant) {
+  $('.Pen-variant-link.active').removeClass('active');
+  $('.Pen-panel.Pen-preview').removeClass('loading');
+  $(clickedVariant).addClass('active');
+  const currentComponent = $('.Frame-panel--main').html();
+  logState(currentComponent, clickedVariant.href);
+}
+
+function logState(currentComponent, url, replace = false) {
```
Comment (AndrewLeedham) [inline assets/js/theme.js]: Not sure log is the right word here?

**Review by AndrewLeedham — CHANGES_REQUESTED** [review]
The tabs look odd when stacked, perhaps it should be horizontally scrollable?
image

**Comment by tinytim84** [comment]
> The tabs look odd when stacked, perhaps it should be horizontally scrollable?
> image

Will it not look strange with a scrollbar underneath it too?  Maybe arrows to the right and left?

**Comment by AndrewLeedham** [comment]
> > The tabs look odd when stacked, perhaps it should be horizontally scrollable?
> > image
> 
> Will it not look strange with a scrollbar underneath it too? Maybe arrows to the right and left?

If you use `overflow-x: auto;` it should be a transparent scrollbar that is what happens with the current framework header bar thingy

**Review by AndrewLeedham — CHANGES_REQUESTED** [review]
Looks good but did something change with the border on the bottom? They are not bleeding into the content anymore: 
image

**Comment by tinytim84** [comment]
> Looks good but did something change with the border on the bottom? They are not bleeding into the content anymore:
> image

It's the scrollbar.  It's almost exactly the same grey as the outline.  If it's not there they still bleed in.

**Comment by AndrewLeedham** [comment]
> It's the scrollbar. It's almost exactly the same grey as the outline. If it's not there they still bleed in.

:( ok lets leave it as this for now, perhaps in the future we can implement something like https://material-ui.com/components/tabs/#automatic-scroll-buttons

**Review by AndrewLeedham — DISMISSED** [review]

**Review by AndrewLeedham — CHANGES_REQUESTED** [review]
Ajax seems somewhat pointless now the tabs disappear while loading. I would say either just use normal links and don't bother with Ajax, or have the loading animation in both the details at the bottom and the display iframe, so the tabs are always visible.

**Comment by tinytim84** [comment]
> Ajax seems somewhat pointless now the tabs disappear while loading. I would say either just use normal links and don't bother with Ajax, or have the loading animation in both the details at the bottom and the display iframe, so the tabs are always visible.

That's a fair point.  I've fixed it in the latest changes.

**Comment by AndrewLeedham** [comment]
LGTM

**Review by AndrewLeedham — APPROVED** [review]

**Pull request merged**